pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -451,6 +450,20 @@ func translateRoute(push *model.PushContext, node *model.Proxy, in *networking.H
 	return out
 }
 
+// translateAppendHeaders translates headers
+func translateAppendHeaders(headers map[string]string) []*core.HeaderValueOption {
+	headerValueOptionList := make([]*core.HeaderValueOption, 0, len(headers))
+	for key, value := range headers {
+		headerValueOptionList = append(headerValueOptionList, &core.HeaderValueOption{
+			Header: &core.HeaderValue{
+				Key:   key,
+				Value: value,
+			},
+		})
+	}
+	return headerValueOptionList
```

Comment:
The ordering of `headerValueOptionList` is going to be platform specific (based on `range headers`). Sort the headers here, so that we have a guaranteed ordering.

Reply:
+1. Otherwise RDS will churn

Reply:
and stablesort

Reply:
Done